backend/de.metas.marketing/serialletter/src/main/java/de/metas/letter/service/async/spi/impl/C_Letter_CreateFromMKTG_ContactPerson_Async.java

```diff
@@ -96,12 +105,23 @@ public Result processWorkPackage(final I_C_Queue_WorkPackage workPackage, final
 				}
 			}
 
+			final BPartnerId bPartnerId = contactPerson.getBPartnerId();
+			final BPartnerLocationId bpLocationId = contactPerson.getBpLocationId();
+			final UserId userId = contactPerson.getUserId();
+
+			final RenderedAddressAndCapturedLocation renderedAddress = documentLocationBL.computeRenderedAddress(
+					DocumentLocation.builder()
+							.bpartnerId(bPartnerId)
+							.bpartnerLocationId(bpLocationId)
+							.contactId(BPartnerContactId.of(bPartnerId,userId))
+							.build());
+
 			final Letter letter = Letter.builder()
 					.boilerPlateId(contactPerson.getBoilerPlateId())
-					.bpartnerId(contactPerson.getBPartnerId())
-					.bpartnerLocationId(contactPerson.getBpLocationId())
-					.userId(contactPerson.getUserId())
-					.address(location.getAddress())
+					.bpartnerId(bPartnerId)
+					.bpartnerLocationId(bpLocationId)
+					.userId(userId)
+					.address(renderedAddress.getRenderedAddress())
```

Comment:
is the `final Location location` still used?

Reply:
@teosarca no, good point! Deleting it!